sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastExchangeExec.scala

```diff
@@ -50,17 +50,20 @@ class SerializeConcatHostBuffersDeserializeBatch(
   extends Serializable with Arm with AutoCloseable {
   @transient private val headers = data.map(_.header)
   @transient private val buffers = data.map(_.buffer)
-  @transient private var batchInternal: ColumnarBatch = null
+  @transient private val batchInternalRef: AtomicReference[ColumnarBatch] =
+    new AtomicReference[ColumnarBatch]()
 
   def batch: ColumnarBatch = {
+    var batchInternal = batchInternalRef.get()
     if (batchInternal == null) {
       // TODO we should come up with a better way for this to happen directly...
```

Comment:
Do we want locking here to prevent multiple threads from wasting time creating duplicate objects?  I'm wondering if things get simpler if we just synchronize the relevant methods.

Reply:
That's a good idea. I will make that change.